### ISSUE TITLE
Setting a config option to false behaves as if set to true

### ISSUE THREAD

**Issue by lorin** [ansible/ansible#3138]
Setting a config option to False (or false, or 0, or no) behaves as if the option was set to True. Example:

```
$ ANSIBLE_ASK_SUDO_PASS=False ansible myhost -m ping -s
sudo password:
```

I believe the config parsing code is missing some logic to convert from strings to boolean.

**Comment by mpdehaan** [comment]
environment variables are a legacy mechanism here, use --ask-sudo-pass

**Comment by lorin** [comment]
This affects all boolean config options, not just environment variable-based things, and not just legacy ones. I used the example above because it was the simplest way to illustrate the problem.

Here's a non-legacy example which I believe will fail, in ansible.cfg

```
[ssh_connection]
scp_if_ssh=False
```

**Comment by mpdehaan** [comment]
I don't think we ever said that was settable via an inventory file, but I don't see why it really shouldn't be.  Yeah, we should fix this.

**Comment by mpdehaan** [comment]
scp_if_ssh I believe is constants.py only though.

**Comment by mpdehaan** [comment]
similarly, some folks want to do -e "x=True" and not have it be a string.

**Comment by bcoca** [comment]
I was going to take a stab at it, but it looks like it was fixed in 9db4f7a9a649a923b88878aa8f04d8394abacff2

**Comment by mpdehaan** [comment]
Yes!
